libraries/core_libs/consensus/src/dag/dag_block_proposer.cpp

```diff
@@ -41,6 +41,22 @@ DagBlockProposer::DagBlockProposer(const FullNodeConfig& config, std::shared_ptr
   auto addr = std::stoull(node_addr.toString().substr(0, 6).c_str(), NULL, 16);
   my_trx_shard_ = addr % bp_config_.shard;
 
+  const auto [dag_gas_limit, pbft_gas_limit] = config.genesis.getGasLimits(final_chain_->lastBlockNumber());
+
+  if (kPbftGasLimit > pbft_gas_limit) {
+    LOG(log_er_) << "Propose pbft gas limit:" << kPbftGasLimit
+                 << " greater than max allowed pbft gas limit:" << pbft_gas_limit;
+    assert(false);
+    return;
+  }
+
+  if (kDagGasLimit > dag_gas_limit) {
+    LOG(log_er_) << "Propose dag gas limit:" << kDagGasLimit
+                 << " greater than max allowed dag gas limit:" << dag_gas_limit;
+    assert(false);
+    return;
+  }
```

Comment:
There is method FullNodeConfig::validate() where we check validity of config values...

Reply:
Moved the check to FullNodeConfig::validate()